docs/src/main/asciidoc/jreleaser.adoc

```diff
@@ -0,0 +1,594 @@
+
+[source,xml]
+----
+    <dependency>
+      <groupId>io.quarkus</groupId>
+      <artifactId>quarkus-picocli</artifactId>
+    </dependency>
+----
+
+== Preparing the project for GitHub releases
+
+The project must be hosted at a GitHub repository before we continue. This task can be completed by logging into your
+GitHub account, creating a new repository, and adding the newly created sources to said repository. Choose the `main`
+branch as default to take advantage of conventions and thus configure less in your `pom.xml`.
+
+You also need a GitHub Personal Access token to be able to post a release to the repository you just created. Follow
+the official documentation for 
+link:https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token[creating a personal access token].
+Store the newly created token at a safe place for future reference. Next, you have the choice of configuring the token
+as an environment variable named `JRELEASER_GITHUB_TOKEN` so that the tool can read it. Alternatively you may store
+the token at a secure location of your choosing, using a `.yml`, `.toml`, `.json`, or `.properties` file. The default
+location is `~/.jreleaser/config[format]`. For example, using the `.yml` format this file could look like
+
+[source,yaml]
+.~/.jreleaser/config.yml
+----
+JRELEASER_GITHUB_TOKEN: <github-token-value>
+----
+
+Alright. Add all sources and create a first commit. You can choose your own conventions for commit messages however you
+can get more bang for your buck when using JReleaser if you follow the 
+link:https://www.conventionalcommits.org/en/v1.0.0/[Conventional Commits] specification. Make your first commit with the
+following message "build: Add initial sources".
+
+== Packaging as a Native Image distribution
+
+Quarkus already knows how to create a native executable using GraalVM Native Image. The default setup will create a
+single executable file following a naming convention. However the JReleaser tool expects a distribution that is, a
+conventional file structure packaged as a Zip or Tar file. The file structure must follow this layout
+
+[source]
+----
+.
+├── LICENSE
+├── README
+└── bin
+    └── executable
+----
+
+This structure lets you add all kinds of supporting files required by the executable, such as configuration files,
+shell completion scripts, man pages, license, readme, and more.
+
+== Creating the distribution
+
+We can leverage the link:http://maven.apache.org/plugins/maven-assembly-plugin/[maven-assembly-plugin] to create such
+a distribution. We'll also make use of the link:https://github.com/trustin/os-maven-plugin[os-maven-plugin] to properly
+identify the platform on which this executable can run, adding said platform to the distribution's filename.
+
+First, let's add the os-maven-plugin to the `pom.xml`. This plugin works as a Maven extension and as such must be added
+to the `<build>` section of the file:
```

Comment:
Maybe put this inside the build section in the example below and add ellipsis to be clear some other things might be there.

Reply:
Applied.